src/Compression/CompressionCodecT64.cpp

```diff
@@ -504,6 +504,12 @@ UInt32 compressData(const char * src, UInt32 bytes_size, char * dst)
     src += bytes_to_skip;
     dst += bytes_to_skip;
 
+    if (bytes_size % sizeof(T) != 0)
+        throw Exception(ErrorCodes::CANNOT_COMPRESS, "Cannot compress with T64 codec, data size {} is not aligned to {}", bytes_size, sizeof(T));
+
+    if (bytes_size == 0)
+        return header_size + bytes_to_skip;
```

Comment:
I think we don't actually write the header in this case.
Makes sense to throw `LOGICAL_ERROR` here, as compressing a buffer of size 0 is definitely not expected.